sda/processor.go

```diff
@@ -32,6 +125,8 @@ func NewServiceProcessor(c *Context) *ServiceProcessor {
 }
 
 // RegisterMessage puts a new message in the message-handler
+// XXX More comments are needed as it's not clear whether RegisterMessage waits
+// for message for/from Clients or for/from Services.
```

Comment:
Well, since you both points it I will do it, I kinda hoped that @ineiti would do it but he's not there so ... :)